internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -435,6 +442,17 @@ func resourceKubernetesClusterNodePoolCreate(d *pluginsdk.ResourceData, meta int
 		return err
 	}
 
+	subnetID, err := commonids.ParseSubnetID(d.Get("vnet_subnet_id").(string))
+	if err != nil {
+		return fmt.Errorf("parsing subnet ID %+v", err)
```

Comment:
The parser functions return plenty information in the error message, so there is no need to wrap them
```suggestion
		return err
```